Upgrade to Spring Boot 1.3

snicoll opened the pull request:
This example can be further simplified by using the  `@JmsListener`
infrastructure. The `MessageConverter` can be easily associated to a
custom factory via a configurer as illustrated in the configuration.

snicoll committed:
Upgrade to Spring Boot 1.3

This example can be further simplified by using the  `@JmsListener`
infrastructure. The `MessageConverter` can be easily associated to a
custom factory via a configurer as illustrated in the configuration.

sbrannen commented:
:+1:

tonvanbart commented:
Thank you for the suggestion - I was aware of the `@JmsListener` annotation, but I have one question about that: in our scenario, destination names need to be configurable via `application.properties` or similar. As far as I've been able to make out, the annotation only takes a String literal, so it seemed to me I couldn't use it. Is there anything I've missed there?
(Also, I found it helpful to visualize for myself all the machinery involved, but that's another story)

snicoll committed:
Demonstrate the use of placeholders

`@JmsListener` (as many others) supports property replacements so all
that is required is to pass the expression to a given value. This commit
complements the story with a custom `ApplicationProperties` that
showcases custom configuration with content assistance in modern IDEs.

snicoll commented:
>  As far as I've been able to make out, the annotation only takes a String literal

What have you tried exactly? If you have `foo.bar` in your `application.properties`, adding `${foo.bar}` is all that's required. I've added another commit that showcases this (that and custom app configuration)

> (Also, I found it helpful to visualize for myself all the machinery involved, but that's another story)

I appreciate that but this is also additional configuration code that you have to maintain. I wouldn't keep that if you don't absolutely need it. Don't forget that boot will back off if it detects certain beans in user configuration: this means that any further improvements we add there may be missed.

tonvanbart commented:
When I try to change my production project to this setup, it fails because the `configurer` bean is not there:
`Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type [org.springframework.boot.autoconfigure.jms.DefaultJmsListenerContainerFactoryConfigurer] found for dependency`
I'm trying to google this error, but so far coming up empty - any suggestions?
